### DISCUSSION TITLE
pre-commit's ansible-lint misses ansible

### DISCUSSION THREAD

**Discussion by GarrisonD** [ansible/ansible-lint#2237] [Q&A]
`pre-commit` clones this repo and runs `pip install .[community]` according to [this](https://github.com/ansible/ansible-lint/blob/8c1cc3457d0a3d96ca74a169638c9fa0354bf30b/.pre-commit-hooks.yaml#L19). After merging [this](https://github.com/ansible/ansible-lint/pull/1888/files#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52L91) there is no `community` extras. As a result, I see failures like `syntax-check: couldn't resolve module/action 'community.docker.docker_login'`.

What about defining a `pre-commit` extra instead of a previously existing `community` one to install Ansible again?

Or is there a better solution?

---

Also, it would be great to add a CI job (I can do that) with a setup for this case.

It took me several hours to understand where the problem is (how `pre-commit` installs hooks, why `ansible-lint` has version `0.1.dev1` instead of `v6.3.0` in venv 🤯 due to some bug in `setuptools-scm` and so on) 😰

**Comment by ssbarnea** [comment]
That's maintenance joy.... Lucky this kind of changes (like removal of an extra) do not appear so often.

**Reply by GarrisonD** [reply]
Is it something that will be fixed soon or is it better to migrate to something else?

**Reply by ssbarnea** [reply]
https://github.com/ansible/ansible-lint/pull/2303 should fix it once merged but you should test it. Be sure that you manually put its revision `382a15a995ba9115ed21abd17040c87748aab486` into your pre-commit hook config. I also included a new pipeline that should ensure that the hook definition remains valid. Hopefully this will address the issue.

One weird thing that I observed is that the hooks was working for me even if community extra was removed long time ago. I suspect this might be a bug in pip which does not fail the installation when the extra does not exist. For this reason I stopped using an extra and I manually mentioned ansible-core nearby.
